src/target-sc/ila_sim.cc

```diff
@@ -149,43 +195,56 @@ void IlaSim::sim_gen_execute_kernel() {
 }
 
 void IlaSim::sim_gen_execute_invoke() {
-  header_ << header_indent_ << "SC_HAS_PROCESS(" << model_ptr_->name() << ");"
-          << endl;
-  header_ << header_indent_ << model_ptr_->name()
-          << "(sc_module_name name_) : sc_module(name_) {" << endl;
-  increase_indent(header_indent_);
-  header_ << header_indent_ << "SC_METHOD(compute);" << endl;
-  header_ << header_indent_ << "sensitive";
-  for (unsigned int i = 0; i < model_ptr_->input_num(); i++) {
-    header_ << " << " << model_ptr_->name() << "_"
-            << model_ptr_->input(i)->name() << "_in";
+  if (!qemu_device_) {
+    header_ << header_indent_ << "SC_HAS_PROCESS(" << model_ptr_->name() << ");"
+            << endl;
+    header_ << header_indent_ << model_ptr_->name()
+            << "(sc_module_name name_) : sc_module(name_) {" << endl;
+    increase_indent(header_indent_);
+    header_ << header_indent_ << "SC_METHOD(compute);" << endl;
+    header_ << header_indent_ << "sensitive";
+    for (int i = 0; i < model_ptr_->input_num(); i++) {
+      header_ << " << " << model_ptr_->name() << "_"
+              << model_ptr_->input(i)->name() << "_in";
+    }
+    header_ << ";" << endl;
+    decrease_indent(header_indent_);
+    header_ << header_indent_ << "}" << endl;
   }
-  header_ << ";" << endl;
-  decrease_indent(header_indent_);
-  header_ << header_indent_ << "}" << endl;
 
   decrease_indent(header_indent_);
   header_ << header_indent_ << "};" << endl;
 }
 
 void IlaSim::sim_gen_export() {
   ofstream outFile;
-  outFile.open(export_dir_ + "test.h");
+  outFile.open(export_dir_ + model_ptr_->name().str() + ".h");
   outFile << header_.rdbuf();
   outFile.close();
-  mk_script_ << "g++ -I. -I " << systemc_path_ << "/include/ "
-             << "-L. -L " << systemc_path_ << "/lib-linux64/ "
-             << "-Wl,-rpath=" << systemc_path_ << "/lib-linux64/ "
-             << "-c -o "
-             << "test_tb.o test_tb.cc "
-             << "-lsystemc" << endl;
-
-  mk_script_ << "g++ -I. -I " << systemc_path_ << "/include/ "
-             << "-L. -L " << systemc_path_ << "/lib-linux64/ "
-             << "-Wl,-rpath=" << systemc_path_ << "/lib-linux64/ "
-             << "-o "
-             << "test_tb test_tb.o " << obj_list_.rdbuf() << "-lsystemc"
-             << endl;
+  if (!qemu_device_) {
+    mk_script_ << "g++ -I. -I " << systemc_path_ << "/include/ "
+               << "-L. -L " << systemc_path_ << "/lib-linux64/ "
+               << "-Wl,-rpath=" << systemc_path_ << "/lib-linux64/ -std=c++11 "
+               << "-c -o "
+               << "uninterpreted_func.o "
+               << "../../uninterpreted_func/uninterpreted_func.cc "
+               << "-lsystemc" << endl;
+    /*
```

Comment:
Unused code should be remove. Otherwise, it should be guarded by defined flags.

Reply:
This comment indicates some fixes for issue#139